src/vs/editor/common/model/textModel.ts

```diff
@@ -1470,9 +1471,15 @@ export class TextModel extends Disposable implements model.ITextModel {
 					0,
 					0
 				));
-				const injectedText = LineInjectedText.fromDecorations(decorationsWithInjectedTextInEditedRange);
+				const injectedTextInEditedRange = LineInjectedText.fromDecorations(decorationsWithInjectedTextInEditedRange);
 
-				const injectedTextFilter = new DecreasingKeyFilter(injectedText, t => t.lineNumber);
+				const decorationsWithInjectedTextInInsertedRange = this._ensureNodesHaveRanges(this._decorationsTree.getInjectedTextInInterval(
```

Comment:
The two ranges used for the calls to `getInjectedTextInInterval`:
* `[(firstEditLineNumber,1) - (lastEditLineNumber, maxColumn(lastEditLineNumber))]` and 
* `[(lastEditLineNumber,1) - (lastInsertedLineNumber,maxColumn(lastInsertedLineNumber))]`
are always overlapping.

I suggest to make a single call to fetch all relevant injected text. i.e. `getInjectedTextInInterval` with the range:
* `[(firstEditLineNumber,1) - (lastInsertedLineNumber,maxColumn(lastInsertedLineNumber)))]`